Release Candidate.

randallli opened the pull request:
### Thanks for starting a pull request on Material Components!

#### Don't forget:
- [ ] Identify the component the PR relates to in brackets in the title. ```[Buttons] Updated documentation```
- [ ] Link to GitHub issues it solves. ```closes #1234```
- [ ] Please add unit tests, snapshot tests, or manual testing steps depending on the change.
- [ ] Sign the CLA bot. You can do this once the pull request is opened.

[Contributing](https://github.com/material-components/material-components-ios/tree/develop/contributing) has more information and tips for a great
pull request.

Commit:
Merge branch 'stable' into release-candidate

Commit:
merged release-candidate

Commit:
Project import generated by Copybara.

PiperOrigin-RevId: 291912424

Commit:
[AppBar] Minor clarification to example documentation.

PiperOrigin-RevId: 291942763

Commit:
Add trackEndsAreRounded property to MDCSlider

PiperOrigin-RevId: 291974805

Commit:
Add trackEndsAreInset property to MDCSlider

PiperOrigin-RevId: 291975646

Commit:
Add allowAnimatedValueChanges property to MDCSlider

PiperOrigin-RevId: 291975707

iberezan committed:
Avoid code duplication for placing action and content scroll views in 'layoutSubviews' method of MDCAlerControllerView.

PiperOrigin-RevId: 291984194

Commit:
Move off of MDCDeviceTopSafeAreaInset (continuation of #9476).

PiperOrigin-RevId: 291998789

Commit:
[kokoro] Add a kokoro auto-labeler GitHub Action.

This will allow our copybara-service worker PRs to automatically run Kokoro jobs.

Closes https://github.com/material-components/material-components-ios/pull/9503

COPYBARA_INTEGRATE_REVIEW=https://github.com/material-components/material-components-ios/pull/9503 from jverkoey:workflow e85fafd1dd75e7cd0b527dc1477d2f9f853941b1
PiperOrigin-RevId: 292008470

iberezan committed:
Ensure to remove correct notification observer for the style updates.

Looks like notification observers are added for MDCAlertController in mdc_setAdjustsFontForContentSizeCategory: method, but never removed.
On the other hand they seem to be removed for MDCAlertControllerView instead, which doesn't look to observe anything.

PiperOrigin-RevId: 292010203

iberezan committed:
Refactor out the layout calculations for title, title icon, message and the view insets into private methods.

This allows to avoid code duplication and enables easier update of the layout to GM2 design.

PiperOrigin-RevId: 292035417

Commit:
Implemented opt-in ability to have a "sticky" view, presented behind top safe area, when a headerViewController is not provided (continuation of #9464).

PiperOrigin-RevId: 292038584

Commit:
Use keyWindow top inset to maintain backwards consistency with behavior prior to moving off of MDCDeviceTopSafeAreaInset.

PiperOrigin-RevId: 292041789

Commit:
Use positive booleans to improve readability of height layout calculations.

PiperOrigin-RevId: 292045083

Commit:
Refactor `CGFloat topAreaInsetForHeader = (self.headerViewController ? self.topSafeAreaInset : 0);` usage into its own private method.

PiperOrigin-RevId: 292048359

yarneo committed:
Add contributor to OWNERS file

PiperOrigin-RevId: 292143787

codeman7 committed:
[TextFields] Add localized clear text string for multiline text field.

PiperOrigin-RevId: 292146071

Commit:
[BottomNavigation] Layout item views after title changes.

When an item's title changes, the Bottom Navigation and its item views should
trigger a layout pass.

Testing
-------
This test cannot be tested easily with Snapshot tests because the Snapshot
test frameworks automatically trigger a layout pass before rendering the view
hierarchy. Testing has to be done manually, most easily through an Example.

1. Set the item's title to an initial value.
2. After the Bottom Navigation is visible, change the title of one of its
   items.

Closes #9519

PiperOrigin-RevId: 292155675

jakerockland committed:
[NavigationDrawer] Update documentation basic examples for MDCBottomDrawerViewController usage.

PiperOrigin-RevId: 292159096

andrewoverton committed:
This change removes some log statements that would otherwise annoy clients.

PiperOrigin-RevId: 292226642

Commit:
[Buttons] Set contentEdgeInsets in theming for MDCFloatingButton with normal shape.

PiperOrigin-RevId: 292240032

iberezan committed:
Keep the title on the screen during scrolling, just like the action buttons, to follow MD guidelines on Dialogs.

When scrolling is required, the dialog title is pinned at the top, with buttons pinned at the bottom. This ensures selected content remains visible alongside the title and buttons, even upon scroll. Dialogs don?t scroll with elements outside of the dialog, such as the background.

PiperOrigin-RevId: 292262396

iberezan committed:
Initialize a few CGFloat expressions in dialogs with a single-precision floating point constant.

Because CGFloat may be either 32 or 64-bit depending on the target, initializing CGFloat expressions with a single-precision floating point constant prevents loosing precision on 64-bit platforms.

Note: Converting to CGFloat is another option. Suggestion here in favor of .f syntax is mostly based on the simplicity.
PiperOrigin-RevId: 292271990

yarneo committed:
[NavigationDrawer] Adding a flag to allow clients to opt-in to when the preferredContentSize of the contentViewController changes, it visually updates the drawer height.

PiperOrigin-RevId: 292303752

yarneo committed:
[NavigationDrawer] Add ability to change the maximumInitialDrawerHeight after drawer has been presented.

PiperOrigin-RevId: 292305019

Commit:
[ProgressView] Allow the progress view to be sized to fractional points.

This allows the progress view to more smoothly represent fractional progress and to take advantage of multi-pixel points on devices with retina displays.

PiperOrigin-RevId: 292342291

iberezan committed:
Add iberezan to the codeowners.

PiperOrigin-RevId: 292358601

Commit:
Revert .gitattributes to the `develop` branch behavior.

This file had previously been pointing to the `stable` branch behavior, but now that we are mirroring out to develop we want to use the `develop` behavior.

This fixes git lfs behavior so that snapshot tests can be committed to pull requests properly again.

PiperOrigin-RevId: 292365273

andrewoverton committed:
This change gives MDCBaseTextField the same default font as UITextField.

PiperOrigin-RevId: 292372227

andrewoverton committed:
This change adds a new property to the private protocol MDCTextControl. This property, labelFrame, is used by the outlined style object during style application. The outlined style object needs to know the label frame because the outline breaks before it hits the label and resumes after it. Accessing `label.frame` has proven unreliable during this scenario because of animations in progress.

PiperOrigin-RevId: 292384279

Commit:
[AppBar] Add a snapshot test for a scrolled app bar.

This test will enable us to validate changes in shadow layer behavior for scrolled app bars.

PiperOrigin-RevId: 292434992

codeman7 committed:
[TextFields] Use correct localization API.

We were using the `NSLocalizedStringWithDefaultValue` API which is supposed to be used with the `genstrings` tool. This updates us to use the correct API which is `localizedStringForKey:value:table:`.

For more context on the `genstrings` tool see: https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/LoadingResources/Strings/Strings.html.

Closes #5616

PiperOrigin-RevId: 292469399

jverkoey committed:
[AppBar] Always apply the default shadow behavior when theming.

Both Primary and Surface theming are expected to have a shadow when scrolled. See https://material.io/components/app-bars-top/#theming for more details.

This fixes a bug where the AppBar's Primary theming implementation would set a shadow layer that would not react to scrolling.

Example before/after of a Surface-themed app bar (note the darker shadow in alignment with the expected behavior)

| Before | After |
|:--|:--|
| ![before](https://user-images.githubusercontent.com/45670/73462821-5e7b4800-434a-11ea-97e9-b6ed8a05696d.png) | ![after](https://user-images.githubusercontent.com/45670/73462834-61763880-434a-11ea-8114-6712e397fe3d.png) |

Related to https://github.com/material-components/material-components-ios/issues/6332

Closes https://github.com/material-components/material-components-ios/pull/9529

COPYBARA_INTEGRATE_REVIEW=https://github.com/material-components/material-components-ios/pull/9529 from jverkoey:theming 62ba0b8d84b204833164745d823002c3f4faf0eb
PiperOrigin-RevId: 292526419

Commit:
[AppBar] Add an AppBar Theming example.

This example provides a toggle between Primary and Surface theming of the AppBar.

PiperOrigin-RevId: 292541295

iberezan committed:
Add a private API for passing a custom title icon view. It can be used for passing a custom animation into the title.

PiperOrigin-RevId: 292824168

Commit:
[FlexibleHeader] Add missing import to MDCFlexibleHeaderView+Private.h

PiperOrigin-RevId: 292926878

galiak11 committed:
[Dialogs] Removing redundant function argument

PiperOrigin-RevId: 293016650

galiak11 committed:
[Dialogs] Adding scuba tests for long, short, text and outlined actions in vertical and horizontal layout, in LTR and RTL.

PiperOrigin-RevId: 293063651

Commit:
Adds availability macro for conditionally compiling OS sensitive code.

PiperOrigin-RevId: 293067907

Commit:
[FlexibleHeader] Add hairline support.

This change introduces two new APIs on MDCFlexibleHeaderViewController: `showsHairline` and `hairlineColor`. Enabling `showsHairline` will show a view fully colored with `hairlineColor` at the bottom of the flexible header view.

This hairline view can be hidden and shown at will, enabling behaviors such as hiding the hairline when the header view's elevation changes (via the setShadowLayer:intensityDidChangeBlock: API on MDCFlexibleHeaderView).

PiperOrigin-RevId: 293140791

andrewoverton committed:
Invalidate intrinsic content size in MDCBaseTextField

This change makes it so that the MDCBaseTextField invalidates its intrinsic content size when either the view's width or the view's layout's calculatedHeight has changed since the intrinsic content size was last calculated.

PiperOrigin-RevId: 293141339

andrewoverton committed:
Remove unwanted animations from filled style.

In certain circumstances when we update the paths of the filled style without animations we need to remove any prior animations.

PiperOrigin-RevId: 293141419

andrewoverton committed:
Reevaluate RTL logic because of internal snapshot testing infrastructure

Some internal snapshot tests force RTL by setting the semantic content attribute. The "is RTL" logic in MDCBaseTextFields up to this point only looked at effective layout direction. This change makes it so that it checks if either RTL or LTR semantic content attributes are being forced before consulting effective layout direction.

PiperOrigin-RevId: 293201200

Commit:
[ButtonBar] Add snapshot tests for title color behavior.

These snapshot tests will be used to verify changes in behavior to the title color of buttons.

PiperOrigin-RevId: 293219347

Commit:
[ButtonBar] Use CGSizeZero instead of CGSizeMake(0, 0).

CGSizeZero is more idiomatic when referring to an empty size.

PiperOrigin-RevId: 293222031

Commit:
internal change

PiperOrigin-RevId: 293283016

andrewoverton committed:
Make padding between title label and detail label smaller

This change gets rid of the dynamic padding between labels in MDCSelfSizingStereoCell. With larger fonts it's too big.

PiperOrigin-RevId: 293427754

andrewoverton committed:
This change makes rippleColor null_resettable on MDCBaseCell

PiperOrigin-RevId: 293430561

codeman7 committed:
[ActionSheet] Change default header to be 8pt tall instead of 0.

When an action sheet header does not have a title or message it should be 8pt tall to give extra spacing to the actions. This was originally made 0 but design has since said this was wrong and should be 8pt.

Closes #8710

PiperOrigin-RevId: 293444072

jakerockland committed:
[NavigationDrawer] Improve swipe-down-to-dismiss behavior for drawers where drawer content is less than `160`. This adjustment ensures that for drawer contents that are quite small (ex. 100pt) the drawer contents to not become undismissable. Without this adjustment it is nearly impossible to achieve a scroll offset of (40pt), the current `kVerticalDistanceDismissalThreshold`, making the drawer effectively undismissable.

PiperOrigin-RevId: 293446935

jakerockland committed:
[NavigationDrawer][A11y] Improve behavior when navigating the bottom drawer with Voice Over. This change makes it possible to navigate back and forth between the `headerViewController` and `contentViewController`, without VO getting stuck, by making the view hierarchy's `accessibilityElements` explicit.

PiperOrigin-RevId: 293533815

Commit:
[ButtonBar] Refactor title color logic to a single location.

The title color logic was previously being applied in three different places in slightly different ways each time. This change consolidates the logic into a single method and invokes this new method from each place where the title color can change; namely:

1. When initially creating the button.
2. When the button's tintColor context changes.
3. When the button item's tintColor property changes.

PiperOrigin-RevId: 293580183

Commit:
[ButtonBar] Assign the UIBarButtonItem's tintColor to the button's text color.

Prior to this change, UIBarButtonItem's tintColor property would not change the title color of the corresponding button. This differs from the UIKit behavior, which does result in the title color being changed.

After this change, changing a UIBarButtonItem's tintColor will result in the corresponding button's title color being updated as well.

PiperOrigin-RevId: 293583337

Commit:
[ButtonBar] Rolling back "Assign the UIBarButtonItem's tintColor to the button's text color."

PiperOrigin-RevId: 293609216

Commit:
[FlexibleHeader] Add missing nullability annotation.

PiperOrigin-RevId: 293672143

galiak11 committed:
Renaming "custom" category to "customize".

PiperOrigin-RevId: 293770541

galiak11 committed:
Renaming "custom" category to "customize".

PiperOrigin-RevId: 293771422

Commit:
[Material][Availability] Migrates Feature Hightlight to MDCAvailability.

PiperOrigin-RevId: 293794429

Commit:
[FlexibleHeader] Add missing self-> within a block.

Resolves the following error:
"Block implicitly retains 'self'; explicitly mention 'self' to indicate this is intended behavior"

PiperOrigin-RevId: 293801599

Commit:
[Availability] Add an umbrella header.

All components are expected, by convention, to have an umbrella header.

PiperOrigin-RevId: 293804134

andrewoverton committed:
Add Swift-friendlier names to TextControl text fields

Automatic Swift header generation worked for the most part. However, stateful setters with the words "LabelColor" in them seemed to not translate correctly. For example...

This method:

-setTrailingAssistiveLabelColor:forState:

Would end up looking like this:

func setTrailingAssistiveLabel(_ trailingAssistiveLabelColor: UIColor, for state: MDCTextControlState)

Which makes it look like it's setting a label, not a label color.

Now those methods look like this:

func setTrailingAssistiveLabelColor(_ trailingAssistiveLabelColor: UIColor, for state: MDCTextControlState)

So when you call it it looks like this:

textField.setTrailingAssistiveLabelColor(.blue, for: .normal)

PiperOrigin-RevId: 293805941

Commit:
[Material][Availability] Migrates elevation to MDCAvailability.

PiperOrigin-RevId: 293812421

Commit:
[FlexibleHeader] Resolve precision warning due to use of floating point.

PiperOrigin-RevId: 293816841

galiak11 committed:
[Dialogs] Update comments describing private margin properties.

PiperOrigin-RevId: 293847981

Commit:
[ButtonBar] Assign the UIBarButtonItem's tintColor to the button's text color.

Prior to this change, UIBarButtonItem's tintColor property would not change the title color of the corresponding button. This differs from the UIKit behavior, which does result in the title color being changed.

After this change, changing a UIBarButtonItem's tintColor will result in the corresponding button's title color being updated as well.

PiperOrigin-RevId: 293848033

galiak11 committed:
[Dialogs] Update comments describing private margin properties.

PiperOrigin-RevId: 293860709

Commit:
[Material][Availability] Migrates the Banner to MDCAvailability.

PiperOrigin-RevId: 293864905

Commit:
[Material][Availability] Migrates Buttons to MDCAvailability.

PiperOrigin-RevId: 293865037

Commit:
Internal Change

PiperOrigin-RevId: 293891746

randallli committed:
[private] Update snapshot tests. (#9666)

Ran snapshot test update to get CI back up and running.

Commit:
[CODEOWNERS] Remove romoore@ (#9660)

Removes romoore@ as a codeowner since he is no longer on the Core team.

randallli committed:
[Availability] add Availability build file (#9678)

randallli committed:
add deps on availability for a bunch of components (#9679)

* add Availability build file

* add Button deps on availability

* added elevation

* corrected include statements to use import of the umbrella files

* revert unwanted changes

* spaces

* clangformat

* add availability dep

* added buttom nav deps on availibility

* Add FeatureHighlight build dep for Availability

* Add Availability build dep to FeatureHighlight private

* Add missing =

* fixed tabs spacing

Co-authored-by: Bryan Oltman <bryanoltman@gmail.com>

bryanoltman committed:
[Buttons] Delete deprecated MDCFloatingActionButtonThemer (#9662)

Deletes the deprecated and unused MDCFloatingActionButtonThemer.

Fixes #9663

randallli committed:
Automatic changelog preparation for release.

randallli committed:
Hand-modified CHANGELOG.md API diff.

randallli committed:
Bumped version number to 103.2.0.

randallli committed:
Hand-modified CHANGELOG.md API diff.

randallli committed:
Hand-modified CHANGELOG.md API diff.

bryanoltman commented on CHANGELOG.md:
```diff
@@ -1,3 +1,153 @@
+Replace this explanations for how to resolve the breaking changes.
+## New deprecations
+
+Replace this text with links to deprecation guides.
```
Complete or delete these sections

yarneo replied:
+1

bryanoltman commented on .github/kokoro_label_all.yml:
```diff
@@ -0,0 +1,21 @@
+# Copyright 2020-present the Material Components for iOS authors. All Rights Reserved.
```
Should this file be in here?

randallli replied:
yea its for copybara-service worker.
https://github.com/material-components/material-components-ios/pull/9503

bryanoltman commented on .github/workflows/kokoro_labeler.yml:
```diff
@@ -0,0 +1,45 @@
+# Copyright 2020-present the Material Components for iOS authors. All Rights Reserved.
```
Same Q about this file

randallli replied:
yep for copybara

bryanoltman commented on CHANGELOG.md:
```diff
@@ -1,3 +1,153 @@
+Bottom drawer added an API for sticky status bars and adjustment when content size changes.
+Slider added an API for allowAnimatedValueChanges and trackEndsAreRounded or trackEndsAreInset.
+
+## API changes
```
This can probably be deleted

bryanoltman replied:
JK, components/Buttons/src/ButtonThemer/MDCFloatingActionButtonThemer.h was deleted

bryanoltman committed:
[TextFields] Update snapshot goldens (#9691)

New golden images were generated using `./.kokoro -d bazel -t //components/TextFields/...` and Xcode 10.3. They're replacing faulty images generated by recording in Xcode.

Partial rollback of https://github.com/material-components/material-components-ios/pull/9666

randallli committed:
Hand-modified CHANGELOG.md API diff.

randallli committed:
Revert "Bumped version number to 103.2.0."

This reverts commit 6663feb418da9b7c8c67e42e64ca65b91b85fcfe.

randallli committed:
modified version number to 104.0.0

yarneo reviewed (approved):
Approving to unblock, but please fill out the changelog with the relevant information about this new release.

yarneo committed:
fix snapshot tests for CI (#9698)

New golden images were generated using ./.kokoro -d bazel -t //components/FeatureHighlight/...  and ./.kokoro -d bazel -t //components/Snackbar/... using Xcode 10.3. They're replacing faulty images generated by recording in Xcode.

randallli committed:
manual edit of changelog

copybara-service[bot] committed:
[FlexibleHeader] Added property for minimumHeaderViewHeight (#9649)

Create new property, minimumHeaderViewHeight, to describe an amount of header to remain on the screen after shiftBehavior is enabled and header has been shifted/scrolled off screen.
